lib/publiccloud/ec2.pm

```diff
@@ -193,7 +194,35 @@ sub upload_img {
 sub terraform_apply {
     my ($self, %args) = @_;
     $args{confidential_compute} = get_var("PUBLIC_CLOUD_CONFIDENTIAL_VM", 0);
-    return $self->SUPER::terraform_apply(%args);
+    my @instances = $self->SUPER::terraform_apply(%args);
```

Comment:
I am confused , do we expect to get array of instances or only one ? seems like you getting an array but all code below is acting like there is only one ...

Reply:
It is usually array containing one member.

Reply:
ok